### PR TITLE
Feature/flowlog

### PR DESCRIPTION
**Pull request by akesser**
### Description of your changes
Support for FlowLogs was added, CloudWatchLogGroup was updated to reflect aws side status

#1269 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Tested manually
[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: apis/ec2/v1alpha1/custom_types.go [hunk 1]
```diff
@@ -449,3 +449,132 @@ type CustomTransitGatewayRouteTableParameters struct {
...
+type CustomFlowLogParameters struct {
+	// Metadata tagging key value pairs
+	// +optional
+	Tags []Tag `json:"tags,omitempty"`
+	// The ID of the subnet, network interface, or VPC for which you want to create
+	// a flow log.
+	//
+	// Constraints: Maximum of 1000 resources
+	// +optional
+	ResourceIDs []*string `json:"resourceIDs"`
```
Comment (haarchri) [inline apis/ec2/v1alpha1/custom_types.go:463]: `json:"resourceIds"`

File: apis/ec2/v1alpha1/custom_types.go [hunk 2]
```diff
@@ -449,3 +449,132 @@ type CustomTransitGatewayRouteTableParameters struct {
...
+	// The type of resource for which to create the flow log. For example, if you
+	// specified a VPC ID for the ResourceId property, specify VPC for this property.
+	// +optional
+	ResourceType *string `json:"resourceType"`
+	// The ID of the VPC for which you want to create a flow log.
+	// +optional
+	// +crossplane:generate:reference:type=github.com/crossplane-contrib/provider-aws/apis/ec2/v1beta1.VPC
+	VPCID *string `json:"vpcID,omitempty"`
```
Comment (haarchri) [inline apis/ec2/v1alpha1/custom_types.go:471]: `json:"vpcId,omitempty"`

File: apis/ec2/v1alpha1/custom_types.go [hunk 3]
```diff
@@ -449,3 +449,132 @@ type CustomTransitGatewayRouteTableParameters struct {
...
+	// VPCIDRef is a reference to an API used to set
+	// the VPCID.
+	// +optional
+	VPCIDRef *xpv1.Reference `json:"vpcIDRef,omitempty"`
```
Comment (haarchri) [inline apis/ec2/v1alpha1/custom_types.go:475]: `json:"vpcIdRef,omitempty"`

File: apis/ec2/v1alpha1/custom_types.go [hunk 4]
```diff
@@ -449,3 +449,132 @@ type CustomTransitGatewayRouteTableParameters struct {
...
+	// VPCIDSelector selects references to API used
+	// to set the VPCID.
+	// +optional
+	VPCIDSelector *xpv1.Selector `json:"vpcIDSelector,omitempty"`
```
Comment (haarchri) [inline apis/ec2/v1alpha1/custom_types.go:479]: `json:"vpcIdSelector,omitempty"`

File: apis/ec2/v1alpha1/custom_types.go [hunk 5]
```diff
@@ -449,3 +449,132 @@ type CustomTransitGatewayRouteTableParameters struct {
...
+	SubnetIDSelector *xpv1.Selector `json:"subnetIdSelector,omitempty"`
+
+	// The ID of the NetworkInterface for which you want to create a flow log.
+	// +optional
+	NetworkInterfaceID *string `json:"networkInterfaceId"`
+
+	// The Amazon Resource Names (ARNs) of an IAM Role.
+	// +crossplane:generate:reference:type=github.com/crossplane-contrib/provider-aws/apis/iam/v1beta1.Role
+	// +crossplane:generate:reference:extractor=github.com/crossplane-contrib/provider-aws/apis/iam/v1beta1.RoleARN()
+	DeliverLogsPermissionARN *string `json:"deliverLogsPermissionARN,omitempty"`
```
Comment (haarchri) [inline apis/ec2/v1alpha1/custom_types.go:533]: `json:"deliverLogsPermissionArn,omitempty"`

File: pkg/controller/cloudfront/lateinit.go [hunk 6]
```diff
@@ -82,7 +84,8 @@ func (fArr filterArr) filter(name string) bool {
 }
 
 // canonicalNameFilter returns a `nameFilter` option that filters all specified canonical CR field names.
-//   Example: `canonicalNameFilter("a.b.c", "a.b.d", "a.b.e")`
+//
+//	Example: `canonicalNameFilter("a.b.c", "a.b.d", "a.b.e")`
```
Comment (haarchri) [inline pkg/controller/cloudfront/lateinit.go:88]: Looks Like to many spaces ?

File: pkg/controller/ec2/flowlog/setup.go [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+			destination := *obj.LogDestination + "/" + *cr.Spec.ForProvider.S3BucketSubfolder + "/"
+			obj.LogDestination = &destination
+		}
+	}
+
+	if cr.Spec.ForProvider.CloudWatchLogDestination != nil {
+		obj.LogDestination = cr.Spec.ForProvider.CloudWatchLogDestination
+	}
+
+	if cr.Spec.ForProvider.Tags != nil {
```
Comment (haarchri) [inline pkg/controller/ec2/flowlog/setup.go:156]: Think we have in other resources the tagger stuff in extra function ?

File: apis/ec2/v1alpha1/custom_types.go (hunk 1 above)
Comment (akesser) [reply apis/ec2/v1alpha1/custom_types.go:463]: Updated the json property

File: apis/ec2/v1alpha1/custom_types.go (hunk 2 above)
Comment (akesser) [reply apis/ec2/v1alpha1/custom_types.go]: Updated the json property

File: apis/ec2/v1alpha1/custom_types.go (hunk 3 above)
Comment (akesser) [reply apis/ec2/v1alpha1/custom_types.go]: Updated the json property

File: apis/ec2/v1alpha1/custom_types.go (hunk 4 above)
Comment (akesser) [reply apis/ec2/v1alpha1/custom_types.go]: Updated the json property

File: apis/ec2/v1alpha1/custom_types.go (hunk 5 above)
Comment (akesser) [reply apis/ec2/v1alpha1/custom_types.go]: Updated the json property

File: pkg/controller/cloudfront/lateinit.go (hunk 6 above)
Comment (akesser) [reply pkg/controller/cloudfront/lateinit.go]: Fixed

File: pkg/controller/ec2/flowlog/setup.go (hunk 7 above)
Comment (akesser) [reply pkg/controller/ec2/flowlog/setup.go:156]: Moved generation of tags in new function generateTagSpecifications

**Review by haarchri — APPROVED** [review]
Thanks for Implementation und improving status outputs

**Pull request merged**